docs/designs/PowerShell-AzF-Overall-Design.md

```diff
@@ -494,7 +494,9 @@ However, if the latest snapshot is _not acceptable_ (i.e. it does not contain mo
 
 When a snapshot installation starts, the dependencies are first installed into a folder with a name following a special pattern (`*.i`), so that this snapshot is not picked up by any worker prematurely, before the installation is complete. After _successful_ completion, the snapshot is _atomically promoted_ by renaming the folder to follow a different pattern (`*.r`), which indicates to other workers that this snapshot is ready to use. If the installation fails or cannot complete for any reason (for example, the worker restarts, crashes, or gets decommissioned), the folder stays in the installing state until removed.
 
-Incomplete and old snapshots that are no longer in use are periodically removed from the file storage. No other changes are ever performed to the snapshots that were once installed.
+Incomplete and old snapshots that are no longer in use are periodically removed from the file storage. In order to allow detecting unused snapshots, each PowerShell worker keeps "touching" a file named `.used` at the root of the used snapshot folder every `PSWorkerHeartbeatPeriodMinutes` minutes. Before and after installing any new snapshot, every PowerShell worker looks for unused snapshots by checking the folder creation time and the `.used` file modification time. If both these time values are older than (`PSWorkerHeartbeatPeriodMinutes` + `PSWorkerOldSnapshotHeartbeatMarginMinutes`) minutes, the snapshot is considered unused, so the PowerShell worker removes it. The latest `PSWorkerMinNumberOfSnapshotsToKeep` snapshots will never be removed, regardless of usage. (`PSWorkerHeartbeatPeriodMinutes`, `PSWorkerOldSnapshotHeartbeatMarginMinutes`, and `PSWorkerMinNumberOfSnapshotsToKeep` are environment variables configurable via Application Settings of a Function App.)
```

Comment:
We should close with @eamonoreilly on the default values for ```PSWorkerMinNumberOfSnapshotsToKeep```, ```PSWorkerHeartbeatPeriodMinutes``` and ```PSWorkerOldSnapshotHeartbeatMarginMinutes```.

Reply:
Yes, and on the names too, as they will be exposed to the user.

Reply:
Since this feature has not been released, we can merge it as it is right now. Then, once we close on the name of the settings, we can update the code.